src/slam_toolbox_multirobot.cpp

```diff
@@ -0,0 +1,254 @@
+
+namespace slam_toolbox
+{
+
+/*****************************************************************************/
+MultiRobotSlamToolbox::MultiRobotSlamToolbox(rclcpp::NodeOptions options)
+: SlamToolbox(options), localized_scan_topic_("/localized_scan")
+/*****************************************************************************/
+{
+    current_ns_ = this->get_namespace() + 1;
+
+    localized_scan_pub_ = this->create_publisher<slam_toolbox::msg::LocalizedLaserScan>(
+    localized_scan_topic_, 10);
+    localized_scan_sub_ = this->create_subscription<slam_toolbox::msg::LocalizedLaserScan>(
+        localized_scan_topic_, 10, std::bind(&MultiRobotSlamToolbox::localizedScanCallback, 
+        this, std::placeholders::_1));
+}
+
+/*****************************************************************************/
+void MultiRobotSlamToolbox::laserCallback(
+  sensor_msgs::msg::LaserScan::ConstSharedPtr scan)
+/*****************************************************************************/
+{
+  // store scan header
+  scan_header = scan->header;
+  // no odom info
+  Pose2 pose;
+  if (!pose_helper_->getOdomPose(pose, scan->header.stamp)) {
+    RCLCPP_WARN(get_logger(), "Failed to compute odom pose");
+    return;
+  }
+
+  // ensure the laser can be used
+  LaserRangeFinder * laser = getLaser(scan);
+
+  if (!laser) {
+    RCLCPP_WARN(get_logger(), "Failed to create laser device for"
+      " %s; discarding scan", scan->header.frame_id.c_str());
+    return;
+  }
+
+  LocalizedRangeScan * range_scan = addScan(laser, scan, pose);
+  if (range_scan != nullptr)
+  {
+    Matrix3 covariance;
+    covariance.SetToIdentity();
+    publishLocalizedScan(scan, laser->GetOffsetPose(), 
+        range_scan->GetOdometricPose(), covariance, scan->header.stamp);
+  }
+}
+
+/*****************************************************************************/
+void MultiRobotSlamToolbox::localizedScanCallback(
+  slam_toolbox::msg::LocalizedLaserScan::ConstSharedPtr localized_scan)
+{
+  std::string scan_ns = localized_scan->scan.header.frame_id.substr(0, 
+                          localized_scan->scan.header.frame_id.find('/'));
+  if (scan_ns == current_ns_) return; // Ignore callbacks from ourself
+
+  sensor_msgs::msg::LaserScan::ConstSharedPtr scan = 
+    std::make_shared<sensor_msgs::msg::LaserScan>(localized_scan->scan);
+  Pose2 pose;
+  pose.SetX(localized_scan->pose.pose.pose.position.x);
+  pose.SetY(localized_scan->pose.pose.pose.position.y);
+  tf2::Quaternion quat_tf;
+  tf2::convert(localized_scan->pose.pose.pose.orientation, quat_tf);
+  pose.SetHeading(tf2::getYaw(quat_tf));
+
+  LaserRangeFinder * laser = getLaser(localized_scan);
+  if (!laser) {
+    RCLCPP_WARN(get_logger(), "Failed to create device for received localizedScanner"
+      " %s; discarding scan", scan->header.frame_id.c_str());
+    return;
+  }
+  LocalizedRangeScan * range_scan = addExternalScan(laser, scan, pose);
+  
+  if (range_scan != nullptr)
```

Comment:
This is also publishing - aren't we having some confliction if this is also publishing?

https://github.com/SteveMacenski/slam_toolbox/blob/ros2/src/slam_toolbox_common.cpp#L472-L498